tests/e2e/utils/versions_test.go

```diff
@@ -0,0 +1,348 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/mod/semver"
+)
+
+type testContext struct {
+	expected    map[string]string
+	sourceEVM   string
+	spacesVM    string
+	sourceAvago string
+	shouldFail  bool
+}
+
+type testMapper struct {
+	app            *application.Avalanche
+	currentContext *testContext
+	srv            *httptest.Server
+	t              *testing.T
+}
+
+func newTestMapper(t *testing.T) *testMapper {
+	app := &application.Avalanche{
+		Downloader: application.NewDownloader(),
+		Log:        logging.NoLog{},
+	}
+	return &testMapper{
+		app,
+		nil,
+		nil,
+		t,
+	}
+}
+
+func (m *testMapper) GetLatestAvagoByProtoVersion(app *application.Avalanche, rpcVersion int, url string) (string, error) {
```

Comment:
why are you passing in a URL?

Reply:
again, I'm not sure if you're comparing against `latest` so you might end up with some prerelease confusion

Reply:
Because the call for the production code (`vm.GetLatestAvalancheGoByProtocolVersion`) requires it.